### PR TITLE
chiplet bus communications with kernel ROM chiplet

### PR DESCRIPTION
**Pull request by grjte**
This PR implements the bus communications for the kernel ROM chiplet, as described [here](https://github.com/0xPolygonMiden/miden-vm/issues/231#issuecomment-1180007490).

- [x] refactor chiplet bus handling to allow multiple requests of different types in the same cycle
- [x] add a kernel rom operation label
- [x] add kernel rom lookups to the chiplets bus
- [ ] update kernel rom tests
- [ ] update documentation

### CODE REVIEW

**Review by bobbinth — COMMENTED** [review]
Looks great! Thank you! And I think the structure is a bit cleaner now too.

I left a few small comment inline. One other question: would the constraints for the `CALL` operation still fit into the limit? I think they should as I think this extra lookup increases the degree only by 1, but would be good to double check.

File: docs/src/design/chiplets/main.md [hunk 1]
```diff
@@ -89,16 +89,17 @@ The labels are composed from the flag values of the chiplet selector(s) and inte
...
 | `HASHER_MR_UPDATE_OLD` |        $\{0\}$        |     $\{1, 1, 0\}$      | $\{0, 1, 1, 0\}$ |   7   |
 | `HASHER_MR_UPDATE_NEW` |        $\{0\}$        |     $\{1, 1, 1\}$      | $\{0, 1, 1, 1\}$ |  15   |
 | `HASHER_RETURN_HASH`   |        $\{0\}$        |     $\{0, 0, 0\}$      | $\{0, 0, 0, 0\}$ |   1   |
 | `HASHER_RETURN_STATE`  |        $\{0\}$        |     $\{0, 0, 1\}$      | $\{0, 0, 0, 1\}$ |   9   |
 | `BITWISE_AND`          |      $\{1, 0\}$       |       $\{0\}$          | $\{1, 0, 0\}$    |   2   |
 | `BITWISE_XOR`          |      $\{1, 0\}$       |       $\{1\}$          | $\{1, 0, 1\}$    |   6   |
-| `MEMORY_READ`          |     $\{1, 1, 0\}$     |       $\{1\}$          | $\{1, 1, 0, 1\}$ |   12  |
+| `MEMORY_READ`          |     $\{1, 1, 0\}$     |       $\{1\}$          | $\{1, 1, 0, 1\}$ |  12   |
 | `MEMORY_WRITE`         |     $\{1, 1, 0\}$     |       $\{0\}$          | $\{1, 1, 0, 0\}$ |   4   |
+| `KERNEL_PROC_CALL`     |     $\{1, 1, 1\}$     |       $\{1\}$          | $\{1, 1, 1, 1}$  |  16   |
```
[the remark this answers is not in the document]
Comment (grjte) [reply docs/src/design/chiplets/main.md:102]: Yep, makes sense. For some reason I thought there was an issue with that, but it looks fine.

**Comment by grjte** [comment]
> Looks great! Thank you! And I think the structure is a bit cleaner now too.
> 
> I left a few small comment inline. One other question: would the constraints for the `CALL` operation still fit into the limit? I think they should as I think this extra lookup increases the degree only by 1, but would be good to double check.

I believe this is fine. I think the only affected constraint is the [block hash computation](https://0xpolygonmiden.github.io/miden-vm/design/decoder/constraints.html#block-hash-computation-constraints) for `SYSCALL` where the degree will increase by 1 due to the kernel ROM procedure lookup. The previous degree was 7.

**Review by bobbinth — APPROVED** [review]
All looks good! Thank you!

**Pull request merged**